ClosedXML/Excel/Ranges/XLRangeBase.cs

```diff
@@ -2249,17 +2249,39 @@ public IXLRangeBase Intersection(IXLRangeBase otherRange, Func<IXLCell, Boolean>
             if (!this.Worksheet.Equals(otherRange.Worksheet))
                 return null;
 
+            Int32 firstRow;
+            Int32 firstColumn;
+
+            Int32 lastRow;
+            Int32 lastColumn;
+
             if (thisRangePredicate == null && otherRangePredicate == null)
             {
                 // Special case, no predicates. We can optimise this a bit then.
-                var range = this.Worksheet.Range
+
+                firstRow = Math.Max
+                (
+                    Math.Min(this.RangeAddress.FirstAddress.RowNumber, this.RangeAddress.LastAddress.RowNumber),
+                    Math.Min(otherRange.RangeAddress.FirstAddress.RowNumber, otherRange.RangeAddress.LastAddress.RowNumber)
+                );
+
+                firstColumn = Math.Max
+                (
+                    Math.Min(this.RangeAddress.FirstAddress.ColumnNumber, this.RangeAddress.LastAddress.ColumnNumber),
+                    Math.Min(otherRange.RangeAddress.FirstAddress.ColumnNumber, otherRange.RangeAddress.LastAddress.ColumnNumber)
+                );
```

Comment:
```
var thisRangeAddress = this.RangeAddress.Normalize();
var otherRangeAddress = otherRange.RangeAddress().Normalize();
var firstRow = Math.Max(thisRangeAddress.FirstAddress.RowNumber, otherRangeAddress.FirstAddress.RowNumber);
var firstColumn = Math.Max(thisRangeAddress.FirstAddress.ColumnNumber, otherRangeAddress.FirstAddress.ColumnNumber);
var lastRow = Math.Min(thisRangeAddress.LastAddress.RowNumber, otherRangeAddress.FirstAddress.RowNumber);
var lastColumn = Math.Min(thisRangeAddress.LastAddress.ColumnNumber, otherRangeAddress.FirstAddress.ColumnNumber);

if (lastRow < firstRow || lastColumn < firstColumn) return null;

return this.Worksheet.Range(firstRow, firstColumn, lastRow, lastColumn);
```

And we don't need `IsNormilized()` method, I suppose.

Reply:
I kind of like the idea of keeping `IsNormalized` around. But I will improve its calculation based on your code.

Reply:
No problem.